pixi.toml

```
@@ -0,0 +1,163 @@
+[project]
+name = "glum"
+channels = ["conda-forge"]
+platforms = ["linux-64", "osx-arm64", "win-64"]
+
+[tasks]
+postinstall = "pip install --no-use-pep517 --no-build-isolation --no-deps --disable-pip-version-check -e ."
+
+[feature.test.tasks]
+test = "pytest tests/glm --doctest-modules src/glum"
+store-golden-master = "python tests/glm/test_golden_master.py"
+store-benchmark-golden-master = "python tests/glm/test_benchmark_golden_master.py"
+
+[feature.docs.tasks]
+make-docs = "cd docs && make html"
+serve-docs = { cmd = "python -m http.server --directory docs/_build/html", depends_on = ["make-docs"] }
+readthedocs = { cmd = "rm -rf $READTHEDOCS_OUTPUT/html && cp -r docs/_build/html $READTHEDOCS_OUTPUT/html", depends_on = ["make-docs"] }
+
+[feature.benchmark.tasks]
+glm-benchmarks-run = "glm_benchmarks_run"
+glm-benchmarks-analyze = "glm_benchmarks_analyze"
+
+[feature.lint.tasks]
+pre-commit-install = "pre-commit install"
+pre-commit-run = "pre-commit run -a"
+
+[feature.no-tabmat.tasks.postinstall-glum-tabmat]
+cmd = """
+pip install --no-use-pep517 --no-build-isolation --no-deps --disable-pip-version-check -e ../tabmat
+&& pip install --no-use-pep517 --no-build-isolation --no-deps --disable-pip-version-check -e .
+"""
+
+[feature.build-tabmat.tasks.install-nightlies]
+cmd = """
+    PRE_WHEELS="https://pypi.anaconda.org/scientific-python-nightly-wheels/simple/"
+    && pip install --pre --no-deps --only-binary :all: --upgrade --timeout=60 -i $PRE_WHEELS pandas
+    && pip install --pre --no-deps --only-binary :all: --upgrade --timeout=60 -i $PRE_WHEELS scikit-learn
+    && pip install --pre --no-deps --only-binary :all: --upgrade --timeout=60 -i $PRE_WHEELS scipy
+    && pip install --pre --no-deps --only-binary :all: --upgrade --timeout=60 -i https://pypi.fury.io/arrow-nightlies/ pyarrow
+    && pip install --no-use-pep517 --no-deps git+https://github.com/Quantco/tabmat
+"""
+
+[build-dependencies]
+c-compiler = "*"
+cxx-compiler = "*"
+cython = "*"
+setuptools-scm = ">=8.1"
+
+[host-dependencies]
+python = ">=3.9"
+pip = "*"
+
+[dependencies]
+numexpr = "*"
+pandas = ">=1.1"
+tabmat = ">=4.0.0"
+scikit-learn = ">=0.23"
+scipy = "*"
+tqdm = "*"
+formulaic = "*"
+pyarrow = "*"
+
+[feature.no-tabmat.dependencies]
+numexpr = "*"
+pandas = ">=1.1"
+scikit-learn = ">=0.23"
+scipy = "*"
+tqdm = "*"
+formulaic = "*"
+pyarrow = "*"
+
+[feature.test.dependencies]
+statsmodels = "*"
+pytest = "*"
+pytest-xdist = "*"
+git_root = "*"
+click = "*"
+attrs = "*"
+psutil = "*"
+mypy = "*"
+
+[feature.dev.dependencies]
+ruff = "*"
+ipdb = "*"
+ipython = "*"
+line_profiler = "*"
+memory_profiler = "*"
```

```suggestion
memory_profiler = "*"
ipykernel = "*"
```
Can we add ipykernel here for vscode users (or should that go somewhere else)?